examples/ambiguity_detection.rs

```diff
@@ -0,0 +1,154 @@
+//! A test to confirm that `bevy` doesn't have system order ambiguity with DefaultPlugins
+//! This is run in CI to ensure that this doesn't regress again.
+
+use bevy::{
+    ecs::schedule::{LogLevel, ScheduleBuildSettings},
+    prelude::*,
+};
+
+/// A test to confirm that `bevy` doesn't have system order ambiguity with DefaultPlugins
+/// This is run in CI to ensure that this doesn't regress again.
+pub fn main() {
+    let mut app = App::new();
+    app.add_plugins(DefaultPlugins);
+    app.edit_schedule(PreStartup, |schedule| {
+        schedule.set_build_settings(ScheduleBuildSettings {
+            ambiguity_detection: LogLevel::Warn,
+            use_shortnames: false,
+            ..default()
+        });
+    });
+    app.edit_schedule(Startup, |schedule| {
+        schedule.set_build_settings(ScheduleBuildSettings {
+            ambiguity_detection: LogLevel::Warn,
+            use_shortnames: false,
+            ..default()
+        });
+    });
+    app.edit_schedule(PostStartup, |schedule| {
+        schedule.set_build_settings(ScheduleBuildSettings {
+            ambiguity_detection: LogLevel::Warn,
+            use_shortnames: false,
+            ..default()
+        });
+    });
+
+    app.edit_schedule(SpawnScene, |schedule| {
+        schedule.set_build_settings(ScheduleBuildSettings {
+            ambiguity_detection: LogLevel::Warn,
+            use_shortnames: false,
+            ..default()
+        });
+    });
+    app.edit_schedule(First, |schedule| {
+        schedule.set_build_settings(ScheduleBuildSettings {
+            ambiguity_detection: LogLevel::Warn,
+            use_shortnames: false,
+            ..default()
+        });
+    });
+    app.edit_schedule(PreUpdate, |schedule| {
+        schedule.set_build_settings(ScheduleBuildSettings {
+            ambiguity_detection: LogLevel::Warn,
+            use_shortnames: false,
+            ..default()
+        });
+    });
+    app.edit_schedule(Update, |schedule| {
+        schedule.set_build_settings(ScheduleBuildSettings {
+            ambiguity_detection: LogLevel::Warn,
+            use_shortnames: false,
+            ..default()
+        });
+    });
+
+    app.edit_schedule(PostUpdate, |schedule| {
+        schedule.set_build_settings(ScheduleBuildSettings {
+            ambiguity_detection: LogLevel::Warn,
+            use_shortnames: false,
+            ..default()
+        });
+    });
+    app.edit_schedule(Last, |schedule| {
+        schedule.set_build_settings(ScheduleBuildSettings {
+            ambiguity_detection: LogLevel::Warn,
+            use_shortnames: false,
+            ..default()
+        });
+    });
+    app.finish();
+    app.cleanup();
+    app.update();
+    assert!(
```

Comment:
We should add logging / reporting of the ambiguities to make this easier to find in CI.

Reply:
They are! But on warn level (and that's why I did my implementation in example because I'm not sure they were displayed with --nocapture ; but yeah displaying all ambiguities in CI is definitely valuable

Reply:
Ah, right! I forgot that it'll automatically log.